ENIGMAsystem/SHELL/Universal_System/Resources/fonts_internal.h

```diff
@@ -40,24 +42,34 @@ namespace enigma
     float xs; // Spacing: used to increment xx
   };
   struct fontglyphrange {
-    fontglyphrange() : glyphstart(0), glyphcount(0) {}
-    unsigned int glyphstart, glyphcount;
+    fontglyphrange() : glyphstart(0) {}
+    unsigned int glyphstart;
     std::vector<fontglyph> glyphs;
   };
-  struct font
+  class font
   {
+    bool destroyed;
+   public:
+     font() : destroyed(false), name(""), fontname(""), fontsize(0),
+       bold(false), italic(false), height(0), yoffset(0), texture(-1),
+       twid(0), thgt(0) {}
     // Trivia
     std::string name, fontname;
-    int fontsize; bool bold, italic;
+    int fontsize; 
+    bool bold, italic;
 
     // Metrics and such
-    unsigned glyphRangeCount;
     std::vector<fontglyphrange> glyphRanges;
     unsigned int height, yoffset;
 
     // Texture layer
     int texture;
     int twid, thgt;
+
+    void destroy() { destroyed = true; }
```

Comment:
Setting a boolean to true is not the same thing as freeing memory. This engine isn't a toy. We don't put on costumes and pretend to do what the API says. We implement real functionality that provides real value. Delete the texture, set it to -1, and *maybe* delete the glyph data.